execution_engine/src/storage/trie_store/db.rs

```diff
@@ -60,12 +68,151 @@ impl<K, V> TrieStore<K, V> for LmdbTrieStore {}
 
 impl<K, V> Store<Digest, Trie<K, V>> for RocksDbStore {
     type Error = error::Error;
-
     type Handle = RocksDb;
-
     fn handle(&self) -> Self::Handle {
         self.rocksdb.clone()
     }
 }
 
 impl<K, V> TrieStore<K, V> for RocksDbStore {}
+
+pub(crate) type Cache = Arc<Mutex<HashMap<Digest, (bool, Trie<Key, StoredValue>)>>>;
+/// In-memory cached trie store, backed by rocksdb.
+#[derive(Clone)]
+pub struct ScratchCache {
+    pub(crate) cache: Cache,
+    pub(crate) rocksdb_store: RocksDbStore,
+}
+
+/// Cached version of the trie store.
+#[derive(Clone)]
+pub struct ScratchTrieStore {
+    pub(crate) store: ScratchCache,
```

Comment:
For the scratch trie store, we can get away with `Arc<Mutex<HashMap<Digest, Trie<Key, StoredValue>>>>`...

Reply:
I opted to newtype over the Arc, but we still need this to pass around as a special associated type in all these operations.